### PR TITLE
fix: sentinel waiting for replication pod got ready

### PR DESCRIPTION
**Pull request by drivebyer**
**Description**

We should wait for the replication cluster to be ready before creating the sentinel cluster.

**Type of change**

* Bug fix (non-breaking change which fixes an issue)

**Checklist**

- [ ] Tests have been added/modified and all tests pass.
- [x] Functionality/bugs have been confirmed to be unchanged or fixed.
- [x] I have performed a self-review of my own code.
- [ ] Documentation has been updated or added where necessary.

**Additional Context**

### CODE REVIEW

File: tests/e2e-chainsaw/v1beta2/teardown/redis-sentinel/sentinel.yaml [hunk 1]
```diff
@@ -8,8 +8,9 @@ spec:
   podSecurityContext:
     runAsUser: 1000
     fsGroup: 1000
-  redisSentinelConfig:
-    redisReplicationName : redis-replication
+   # If we specify the redisReplicationName, we should wait for the redis cluster to be ready before deploying the sentinel
```
Comment (shubham-cmyk) [inline tests/e2e-chainsaw/v1beta2/teardown/redis-sentinel/sentinel.yaml:11]: Why don't we use the chinsaw wait operation in another step so that we can wait till the redis replication is ready @drivebyer

Comment (drivebyer) [reply tests/e2e-chainsaw/v1beta2/teardown/redis-sentinel/sentinel.yaml:11]: > Why don't we use the chainsaw wait operation in another step so that we can wait till the redis replication is ready @drivebyer

In my opinion, in this test case, it is sufficient to install Sentinel independently, and then test the StatefulSet and Service. We do not specify redisReplicationName, the Sentinel StatefulSet and Service will still be created as well.

Comment (drivebyer) [reply tests/e2e-chainsaw/v1beta2/teardown/redis-sentinel/sentinel.yaml:11]: We have already test RedisReplication with RedisSentinel in https://github.com/OT-CONTAINER-KIT/redis-operator/tree/master/tests/e2e-chainsaw/v1beta2/ha-setup :) @shubham-cmyk
